File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/course_detail/uploads/view/CourseUploadsRecyclerAdapter.java
package org.wikiedufoundation.wikiedudashboard.course_detail.uploads.view;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import org.wikiedufoundation.wikiedudashboard.R;
import org.wikiedufoundation.wikiedudashboard.course_detail.uploads.data.CourseUpload;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class CourseUploadsRecyclerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private Context context;
    private LayoutInflater layoutInflater;
    private List<CourseUpload> courseUploads = new ArrayList<>();
    private CourseUpload courseUpload;

    CourseUploadsFragment courseUploadsFragment;

    CourseUploadsRecyclerAdapter(Context context, CourseUploadsFragment courseUploadsFragment) {
        this.context = context;
        layoutInflater = LayoutInflater.from(context);
        this.courseUploadsFragment = courseUploadsFragment;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view1 = layoutInflater.inflate(R.layout.item_rv_course_upload, parent, false);
        return new MyDashboardViewHolder(view1);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, final int position) {
        courseUpload = courseUploads.get(position);
        final MyDashboardViewHolder myDashboardViewHolder = (MyDashboardViewHolder) holder;
        myDashboardViewHolder.tv_upload_title.setText(courseUpload.getFile_name());
        Glide.with(context).load(courseUpload.getThumbUrl()).into(myDashboardViewHolder.iv_course_upload);
        holder.itemView.setOnClickListener(view -> {
//            courseUploadsFragment.openCourseDetail(courseUploads.get(position).getSlug());
        });
    }


    void setData(List<CourseUpload> courseUploads) {
        this.courseUploads = courseUploads;
    }

    @Override
    public int getItemCount() {
        return courseUploads.size();
    }

    public class MyDashboardViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.tv_upload_title)
        TextView tv_upload_title;

        @BindView(R.id.iv_course_upload)
        ImageView iv_course_upload;

        public MyDashboardViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/adapters/CourseUploadsRecyclerAdapter.kt
package org.wikiedufoundation.wikiedudashboard.adapters

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide

import org.wikiedufoundation.wikiedudashboard.R
import org.wikiedufoundation.wikiedudashboard.course_detail.uploads.data.CourseUpload
import org.wikiedufoundation.wikiedudashboard.course_detail.uploads.view.CourseUploadsFragment

import java.util.ArrayList

import kotlinx.android.synthetic.main.item_rv_course_upload.view.*

class CourseUploadsRecyclerAdapter(private val context: Context, internal var courseUploadsFragment: CourseUploadsFragment) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    private var courseUploads: List<CourseUpload> = ArrayList()
    private var courseUpload: CourseUpload? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view1 = LayoutInflater.from(context).inflate(R.layout.item_rv_course_upload, parent, false)
        return MyDashboardViewHolder(view1)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        courseUpload = courseUploads[position]
        val myDashboardViewHolder = holder as MyDashboardViewHolder
        myDashboardViewHolder.tvUploadTitle.text = courseUpload!!.file_name
        Glide.with(context).load(courseUpload!!.thumbUrl).into(myDashboardViewHolder.ivCourseUpload)
        holder.itemView.setOnClickListener {
            //            courseUploadsFragment.openCourseDetail(courseUploads.get(position).getSlug());
        }
    }


    fun setData(courseUploads: List<CourseUpload>) {
        this.courseUploads = courseUploads
    }

    override fun getItemCount(): Int {
        return courseUploads.size
    }

    inner class MyDashboardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvUploadTitle: TextView = itemView.tv_upload_title
        val ivCourseUpload: ImageView = itemView.iv_course_upload
    }
}